lib/statement/where.js

```diff
@@ -143,6 +172,13 @@ const generateConjunction = (conditions, offset, generator) => {
       conjunction.params = conjunction.params.concat(disjunction.params);
       conjunction.predicates.push(`(${disjunction.predicates.join(' OR ')})`);
 
+      return conjunction;
+    } else if (key === 'and') {
+      const inner_conjunction = generateInnerConjunction(value, conjunction.offset + conjunction.params.length, generator);
```

Comment:
I think you should be able to recurse by calling `generateConjunction` directly and appending the new `params` and `predicates` as with the `generateDisjunction` invocation immediately above; from what I can see, directing it through `generateInnerConjunction` just adds another stack frame.

Reply:
You'd have to change this to take an array, the point of generateInnerConjunction is that it takes an array of criteria instead of one criteria. You could essentially paste the body of generateInnerConjunction in here by moving the _.reduce call here, but the same argument is true for generateDisjuntction...

Reply:
oh that's right, my bad :woman_facepalming: